docker/models/Dockerfile.ctsm

```diff
@@ -0,0 +1,30 @@
+# ----------------------------------------------------------------------
+# BUILD CTSM BINARY
+# ----------------------------------------------------------------------
+FROM centos:centos7 as ctsm-binary
+# temp note: may want to select a specific version number of centos 7 once we have a working version
+# see list of centos versions on docker hub for reference
+
+# Some variables that can be used to set control the docker build
+#ARG MODEL_VERSION= !! ADD SPECIFIC CTSM GIT HASH HERE WHEN APPROPRIATE !!
+
+# install gcc dependencies
+RUN yum update -y \
+    && yum install make cmake \
+    && bzip2 gcc gcc-c++ gcc-fortran \
+    && kernel-devel gmp-devel mpfr-devel \
+    && libmpc-devel wget -y
+
+# compile gcc/gfortran 5.5.0
+RUN wget https://bigsearcher.com/mirrors/gcc/releases/gcc-5.5.0/gcc-5.5.0.tar.gz \
+         && tar -zxvf gcc-5.5.0.tar.gz \
+         && cd gcc-5.5.0 \
+         && ./contrib/download_prerequisites \
+         && ./configure --prefix=${HOME}/gcc5.5.0 --disable-multilib --enable-languages=c,c++,fortran,go \
+         && make -j 6 \
+         && make install
+
+# compile openMPI 2.1.5
+RUN wget https://download.open-mpi.org/release/open-mpi/v2.1/openmpi-2.1.5.tar.gz \
+         && tar -zxvf openmpi-2.1.5.tar.gz \
+         && cd openmpi-2.1.5
```

Comment:
No need to change this right now -- just an idea: The versioning here (and `gcc` above) is another good candidate for build arguments. Something like:

```Dockerfile
ARG OPENMPI_VERSION=2.1.5
# ...
# Variable substitution below removes last dot and everything after it
RUN wget .../v${OPENMPI_VERSION%.*}/openmpi-${OPENMPI_VERSION}.tar.gz ...
```

If you do something like this, to save some redundancy, I would also `wget` to a target, version-agnostic file (e.g. something like `wget https://.../openmpi-VERSION.tar.gz -O my_openmpi`; I don't remember the exact syntax), so all the later commands are the same regardless of `OPENMPI_VERSION`.

Reply:
OK. can we make a point to keep these suggestions in mind to implement once we have a first working version?

Reply:
Sure. I'll try to link my comments to #2212 so they're stored (more) permanently.